Add MPAS App to Chiltepin

christopherwharrop-noaa opened the pull request:
This PR adds a self-contained MPAS app to demonstrate Chiltepin workflows for a "real" NWP model.  The App is self-contained such that, given a proper spack-stack or other supporting stack, it will download, build, and install all necessary modeling components, fetch input data and generate grid mesh files, and run all steps from ingesting the GFS grib files to the forecast.  Closes #96

christopherwharrop-noaa commented:
@NaureenBharwaniNOAA - This is ready for your review

NaureenBharwaniNOAA reviewed (commented):
Here's my initial review.

christopherwharrop-noaa commented on requirements.txt:
```diff
@@ -2,22 +2,21 @@ pytest
 jinja2==3.1.*
-jsonschema==4.22.*
-lxml==5.2.*
+jsonschema==4.21.*
```
Yes.  This matches uwtools requirements.

christopherwharrop-noaa commented on src/chiltepin/utils/data_locations.yml:
It is copied verbatim from the mpas_app.  And I'm not going to change it unless the mpas app changes it.

christopherwharrop-noaa commented on src/chiltepin/mpas/wrapper.py:
```diff
@@ -0,0 +1,284 @@
+            stdout=None,
+            stderr=None,
+        ):
+            clone_task = self.get_clone_task(executors=clone_executors)
+            make_task = self.get_make_task(executors=make_executors)
```
That won't work and isn't appropriate here.

christopherwharrop-noaa commented on apps/mpas/config/user_config.yaml:
```diff
@@ -0,0 +1,4 @@
+user:
+  experiment_dir: /home/admin/work/apps/mpas/experiments/test
+  platform: chiltepin
+  resolution: 240 
```
i just changed it to 480 to make it run as fast as possible. That is the coarsest possible resolution.

christopherwharrop-noaa commented on src/chiltepin/utils/retrieve_data.py:
This code is copied verbatim from mpas_app.  If they change it, I'll change it.

christopherwharrop-noaa commented:
@NaureenBharwaniNOAA - I changed my mind about the data retrieval documentation.  I added some notes at the top of the python code and the corresponding config file to explain its provenance.  However, I did not alter the original documentation inside.  Hopefully that helps to address your comments on those.

NaureenBharwaniNOAA reviewed (approved):
I really appreciate the cleanup from `chiltepin` and `ci` to `docker`. It helps clarify things quite a bit. LGTM!
